Library/Homebrew/rubocops/formula_desc_cop.rb

```diff
@@ -0,0 +1,84 @@
+require_relative "./extend/formula_cop"
+require_relative "../extend/string"
+
+module RuboCop
+  module Cop
+    module Homebrew
+      # This cop audits `desc` in Formulae
+      #
+      # - Checks for existence of `desc`
+      # - Checks if size of `desc` > 80
+      # - Checks if `desc` begins with an article
+      # - Checks for correct usage of `command-line` in `desc`
+      # - Checks if `desc` contains the formula name
+
+      class FormulaDesc < FormulaCop
+        attr_accessor :formula_name, :description, :source_buffer, :line_number, :line_begin_pos,
+                      :desc_begin_pos, :call_node
+
+        def audit_formula(node, class_node, _parent_class_node, body)
+          check(node, body, class_node.const_name)
+        end
+
+        private
+
+        def check(node, body, formula_name)
+          body.each_child_node(:send) do |call_node|
+            _receiver, call_name, args = *call_node
+            next if call_name != :desc || args.children[0].empty?
+            @formula_name = formula_name
+            @description = args.children[0]
+            @source_buffer = call_node.source_range.source_buffer
+            @line_number = call_node.loc.line
+            @line_begin_pos = call_node.source_range.source_buffer.line_range(call_node.loc.line).begin_pos
+            @desc_begin_pos = call_node.children[2].source_range.begin_pos
+            @call_node = call_node
```

Comment:
I'm thinking some of this could be extracted into a method named `find_method_by_name(:desc)` which then returns the node or perhaps just the parameters? That's the type of thing I'm thinking with splitting this into smaller methods; have generic functions like `find_method_by_name` and later when needed `find_block_by_name` etc. which could take method `name` and optional `block_name` arguments and return e.g. the parameters or information as a hash. This should all live in `FormulaCop` so that it's not necessary to really understand RuboCop to implement a `FormulaCop` subclass and you can just call these documented, public methods. What do you think?